src/upgrade/steps/consumer-signal.js

```
@@ -0,0 +1,51 @@
+    return pkg.version || 'unknown';
+  } catch {
+    return 'unknown';
+  }
+}
+
+async function plan() {
+  const signalDir = path.join(getAutopgRoot(), 'state');
+  return `would write upgrade signal at ${path.join(signalDir, 'upgrade.signal')}`;
+}
+
+async function execute({ log }) {
```

ESLint

[error] 'log' is defined but never used. Allowed unused args must match /^_/u.

(unused-imports/no-unused-vars)